src/codegen/barrier.cpp

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// barrier.cpp
+//
+// Identification: src/codegen/barrier.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "common/logger.h"
+#include "codegen/barrier.h"
+#include "codegen/utils/oa_hash_table.h"
+
+namespace peloton {
+namespace codegen {
+
+void Barrier::InitInstance(Barrier *ins, uint64_t n_workers)
```

Comment:
Interesting C++ detail: Note that you create a `Barrier` on the stack (on LLVM level) by first allocate memory on the stack, and then call this `InitInstance()` function; and delete a `Barrier` by calling `Destroy()`. You might want to change these 2 functions into using placement new/delete, then you could write normal constructor/destructor.